cirq/protocols/pauli_expansion.py

```diff
@@ -32,9 +33,9 @@ def _filter_coefficients(expansion: Dict[str, complex],
 def pauli_expansion(
     val: Any,
     *,
-    default: Optional[Dict[str, complex]] = RaiseTypeErrorIfNotProvided,
+    default: Union[Dict[str, complex], TDefault] = RaiseTypeErrorIfNotProvided,
```

Comment:
This doesn't trigger the "mutable default value" warning?

In other cases I had to specify overloads with and without the default value in order to get mypy to understand.

Reply:
Re-checked and mypy is happy with this.

Reply:
Re-run pip install with dev reqs to make sure I'm using correct mypy version. Mypy is still happy. As is the one run by travis, so I'm going to keep as-is.